### PR TITLE
Viewer loading progress bar

### PR DESCRIPTION
**Pull request by ryantrem**
This PR adds a default loading progress bar. The styling can be customized (via CSS variables for things like color or the progress-bar part for things like height, border radius, etc.) or completely replaced with custom UI (via the progress-bar slot). For cases where we don't know the size of the download, it is a simple indeterminate progress bar.

Currently this is only for model source download, it does not take into account environment texture download. I think we should probably bring this in as well, but it requires more changes to core. Open to thoughts though.

https://github.com/user-attachments/assets/daf9bd24-670e-4c00-99f4-69c1803b63e5

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/15784/merge

Test environment: 
https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15784/merge/index.html

To test a playground add it to the URL, for example:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15784/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/15784/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/15784/merge
https://gui.babylonjs.com/?snapshot=refs/pull/15784/merge
https://nme.babylonjs.com/?snapshot=refs/pull/15784/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/15784/merge#BCU1XR#0

**Comment by bjsplat** [comment]
WebGL2 visualization test reporter:

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15784/merge/testResults/webgl2playwright/index.html

**Comment by bjsplat** [comment]
Visualization tests for WebGPU (Experimental)

https://snapshots-cvgtc2eugrd3cgfd.z01.azurefd.net/refs/pull/15784/merge/testResults/webgpuplaywright/index.html

File: packages/tools/viewer-alpha/src/viewerElement.ts [hunk 1]
```diff
@@ -170,20 +171,72 @@ export class HTML3DElement extends LitElement {
...
+                width: 0%;
+            }
+            20% {
+                left: 0%;
+                width: 30%;
+            }
+            80% {
+                left: 70%;
+                width: 30%;
+            }
```
Comment (RaananW) [inline packages/tools/viewer-alpha/src/viewerElement.ts:220]: i just wonder why it doesn't match the actual % of the keyframe?

File: packages/tools/viewer-alpha/src/viewer.ts [hunk 2]
```diff
@@ -190,6 +191,11 @@ export class Viewer implements IDisposable {
      */
     public readonly onModelError = new Observable<unknown>();
 
+    /**
+     * Fired when progress changes on loading activity.
+     */
+    public readonly onLoadingProgressChanged = new Observable<void>();
```
Comment (RaananW) [inline packages/tools/viewer-alpha/src/viewer.ts:197]: wouldn't it be better for developers if the observable notified the actual progress and not just void?

**Review by RaananW — APPROVED** [review]

File: packages/tools/viewer-alpha/src/viewer.ts (hunk 2 above)
Comment (ryantrem) [reply packages/tools/viewer-alpha/src/viewer.ts:197]: Talked offline, will leave it as is for now.

File: packages/tools/viewer-alpha/src/viewerElement.ts (hunk 1 above)
Comment (ryantrem) [reply packages/tools/viewer-alpha/src/viewerElement.ts]: Math!

**Pull request merged**